### ISSUE TITLE
Error Handling in cli tool

### ISSUE THREAD

**Issue by rads284** [VowpalWabbit/vowpal_wabbit#2290]
#### Describe the bug
The vw cli tool does not check for bad parameters. It doesn’t validate the command being passed.

#### To Reproduce
1. Used this specific command line i.e. `vw -`

#### Expected behavior
Bad parameter and help should appear.

#### Observed Behavior
`Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = -
can't open '-', sailing on!
num sources = 0
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features

finished run
number of examples = 0
weighted example sum = 0.000000
weighted label sum = 0.000000
average loss = n.a.
total feature number = 0`

#### Environment
What version of VW did you use?
8.8.0

**Comment by jackgerrits** [comment]
There is a single positional parameter that is the data file to be used. In this case the tool is interpreting the `-` as the datafile as you can see by these lines:
 ```
Reading datafile = -
can't open '-', sailing on~
```

**Comment by RituRajSingh878** [comment]
By using only this `vw` command, same behavior is observed also it hangs in end. So I think it should check the passing argument and with the bad parameter, it should suggest to use help.

**Comment by jackgerrits** [comment]
When running just `vw`, it is waiting to accept input from stdin. If you pass EOF (cntrl+D) then it will exit gracefully.

**Comment by RituRajSingh878** [comment]
@jackgerrits thanks
my bad!

**Comment by rads284** [comment]
Shouldn't it hold of other outputs as well?

**Comment by jackgerrits** [comment]
Could you elaborate? I’m not sure I understand.  Are you noticing VW failing to error out on other invalid command lines?

**Comment by rads284** [comment]
No, I am not. My bad.

**Comment by jackgerrits** [comment]
Closing as it seems as though this is resolved, if there are further issues please reopen.
